react-components/src/components/RevealToolbar/LayersButton.tsx

```diff
@@ -2,29 +2,121 @@
  * Copyright 2023 Cognite AS
  */
 
-import { type ReactElement, useState } from 'react';
+import React, { type ReactElement, useState, useEffect, useMemo, useRef } from 'react';
 import { Button, Dropdown } from '@cognite/cogs.js';
-import { LayersContainer } from './LayersContainer/LayersContainer';
+import { type Reveal3DResourcesStates } from './LayersContainer/types';
+import LayersContainer from './LayersContainer/LayersContainer';
+import {
+  type CognitePointCloudModel,
+  type CogniteCadModel,
+  type CogniteModel
+} from '@cognite/reveal';
+import { useReveal } from '../RevealContainer/RevealContext';
+import { use3DModelName } from '../../hooks/use3DModelName';
+import isEqual from 'lodash/isEqual';
 
 export const LayersButton = (): ReactElement => {
```

Comment:
Try to refactor this component into custom hooks / local functions / lambdas. Very hard to follow the logic here.